SandboxTools/SandboxToolsPatches.cs

```diff
@@ -76,9 +77,20 @@ public sealed class SandboxToolsPatches : KMod.UserMod2 {
 			filters.Add(new SearchFilter(SandboxToolsStrings.FILTER_GEYSERS,
 				(entity) => {
 					var prefab = entity as KPrefabID;
-					return prefab != null && (prefab.TryGetComponent(out Geyser _) || prefab.
-						PrefabTag.Name == OilWellConfig.ID);
+					return prefab != null && ((prefab.TryGetComponent(out Geyser _) && prefab.TryGetComponent(out Uncoverable _))
+					|| prefab.PrefabTag.Name == OilWellConfig.ID);
 				}, null, Def.GetUISprite(Assets.GetPrefab("GeyserGeneric_slush_water"))));
+			// Rover
+			if (DlcManager.IsExpansion1Active()) {
+				var rover_name = GameTags.Robots.Models.ScoutRover.Name.ToUpper();
+				var icon = new Tuple<Sprite, Color>(CodexCache.entries[rover_name].icon, CodexCache.entries[rover_name].iconColor);
```

Comment:
`CodexCache.entries[rover_name]` should be stored to a variable and null checked before use